sig/datadog/di/probe_notification_builder.rbs

```diff
@@ -0,0 +1,36 @@
+module Datadog
+  module DI
+    class ProbeNotificationBuilder
+      @serializer: Serializer
+
+      def initialize: (untyped settings, Serializer serializer) -> void
+
+      attr_reader settings: untyped
+      attr_reader serializer: Serializer
+
+      def build_received: (Probe probe) -> untyped
+
+      def build_installed: (Probe probe) -> untyped
+
+      def build_emitting: (Probe probe) -> untyped
+
+      def build_executed: (Probe probe, ?trace_point: untyped?, ?rv: untyped?, ?duration: untyped?, ?callers: untyped?, ?args: untyped?, ?kwargs: untyped?, ?serialized_entry_args: untyped?) -> untyped
+
+      def build_snapshot: (Probe probe, ?rv: untyped?, ?snapshot: untyped?, ?duration: untyped?, ?callers: untyped?, ?args: untyped?, ?kwargs: untyped?, ?serialized_entry_args: untyped?) -> { service: untyped, :"debugger.snapshot" => { id: untyped, timestamp: untyped, evaluationErrors: ::Array[untyped], probe: { id: untyped, version: 0, location: untyped }, language: "ruby", stack: untyped, captures: untyped }, duration: untyped, host: nil, logger: { name: untyped, method: untyped, thread_name: untyped, thread_id: untyped, version: 2 }, :"dd.trace_id" => 136035165280417366521542318182735500431, :"dd.span_id" => 17576285113343575026, ddsource: "dd_debugger", message: untyped, timestamp: untyped }
```

Comment:
Minor: This return looks quite hard to maintain as-is. Perhaps worth either generalizing (into just `Hash`) or introducing a class for this?

Reply:
Repaired.